pkg/operator/hive/hive.go

```diff
@@ -152,6 +161,96 @@ func (r *ReconcileHiveConfig) deployHive(hLog log.FieldLogger, h *resource.Helpe
 		hiveDeployment.Spec.Replicas = &replicas
 	}
 
+	hiveControllersConfigMap := &corev1.ConfigMap{}
+	switch err := r.Get(context.TODO(), types.NamespacedName{Namespace: utils.GetHiveNamespace(), Name: HiveControllersConfigMapName}, hiveControllersConfigMap); {
+	case apierrors.IsNotFound(err):
+		hLog.Infof("configmap for hive controllers not found, creating a new one")
+		newHiveControllersConfigMap := &corev1.ConfigMap{}
+		newHiveControllersConfigMap.Name = HiveControllersConfigMapName
+		newHiveControllersConfigMap.Namespace = utils.GetHiveNamespace()
+		newHiveControllersConfigMap.Data = make(map[string]string)
+		hiveControllersConfigMap = newHiveControllersConfigMap
+		if err := r.Create(context.TODO(), hiveControllersConfigMap); err != nil {
```

Comment:
Do not create it here only to update it later. Do the create after the configmap has been filled out.

Consider using `util.ApplyRuntimeObjectWithGC`, since we also want an owner reference on the configmap.